static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx

```diff
@@ -0,0 +1,257 @@
+    [SpanMetricsFields.SPAN_GROUP]: string;
+  };
+
+  const {isLoading: areSpanMetricsSeriesLoading, data: spanMetricsSeriesData} =
+    useSpanMetricsSeries(
+      groupId,
+      queryFilter,
+      [`avg(${SpanMetricsFields.SPAN_SELF_TIME})`, 'spm()', 'http_error_count()'],
+      'api.starfish.span-summary-page-metrics-chart'
+    );
+
+  useSynchronizeCharts([!areSpanMetricsSeriesLoading]);
+
+  const spanMetricsThroughputSeries = {
+    seriesName: span?.[SpanMetricsFields.SPAN_OP]?.startsWith('db')
+      ? 'Queries'
+      : 'Requests',
+    data: spanMetricsSeriesData?.['spm()'].data,
+  };
+
+  const spanOperationDescription = getSpanOperationDescription(
+    span[SpanMetricsFields.SPAN_OP]
+  );
+
+  return (
+    <React.Fragment>
+      <HeaderContainer>
+        <FilterOptionsContainer>
+          <StarfishDatePicker />
+        </FilterOptionsContainer>
+
+        <BlockContainer>
+          {span?.[SpanMetricsFields.SPAN_OP]?.startsWith('db') &&
+            span?.[SpanMetricsFields.SPAN_OP] !== 'db.redis' && (
+              <Block title={t('Table')}>{span?.[SpanMetricsFields.SPAN_DOMAIN]}</Block>
+            )}
+
+          <Block
+            title={getThroughputTitle(span?.[SpanMetricsFields.SPAN_OP])}
+            description={tct('Throughput of this [spanType] per minute', {
+              spanType: spanOperationDescription,
+            })}
+          >
+            <ThroughputCell rate={spanMetrics?.['spm()']} unit={RateUnits.PER_MINUTE} />
+          </Block>
+
+          <Block
+            title={DataTitles.avg}
+            description={tct(
+              'The average duration of [spanType] in the selected period',
+              {
+                spanType: spanOperationDescription.endsWith('y')
+                  ? `${spanOperationDescription.slice(0, -1)}ies`
+                  : `${spanOperationDescription}s`,
+              }
+            )}
+          >
+            <DurationCell
+              milliseconds={spanMetrics?.[`avg(${SpanMetricsFields.SPAN_SELF_TIME})`]}
+            />
+          </Block>
+
+          {span?.[SpanMetricsFields.SPAN_OP]?.startsWith('http') && (
+            <Block
+              title={t('5XX Responses')}
+              description={t('5XX responses in this span')}
+            >
+              <CountCell count={spanMetrics?.[`http_error_count()`]} />
+            </Block>
+          )}
+
+          <Block
+            title={t('Time Spent')}
+            description={t(
+              'Time spent in this span as a proportion of total application time'
+            )}
+          >
+            <TimeSpentCell
+              percentage={spanMetrics?.['time_spent_percentage()']}
+              total={spanMetrics?.[`sum(${SpanMetricsFields.SPAN_SELF_TIME})`]}
+            />
+          </Block>
+        </BlockContainer>
+      </HeaderContainer>
+
+      {span?.[SpanMetricsFields.SPAN_DESCRIPTION] && (
+        <DescriptionContainer>
+          <SpanDescription
+            span={{
```

Comment:
I wonder if `SpanDescription` should even get access to the entire span - it really only cares about a description string.

Reply:
It definitely should not. It cares about description and op, and it should take them separately. The typing specifies that it only needs a subset, but that's not convenient or clear!